src/app/pages/index.component.ts

```diff
@@ -15,23 +15,37 @@ export class IndexComponent {
   searchValue: string = '';
   showAddCourse: boolean = false;
   _mainThread: Subscription;
+  page = {current: 0, total: 0};
 
   constructor(private _service: CourseService) {}
 
   ngOnInit() {
     this._mainThread = this._service.getList()
-                .map(list => list.filter(course => Number(course.date) > new Date().getTime() - 14 * 24 * 60 * 1000))
-                .subscribe(courses => this.courses = courses);
+                .map(res => {
+                  if(res.facets && res.facets.pages) {
+                    this.page = res.facets.pages;
+                  }
+                  let list = res.courses ? res.courses.filter(item => new Date(item.date).getTime() > new Date().getTime() - 14 * 24 * 60 * 60 * 1000): [];
+                  return list;
+                })
```

Comment:
it's bad to use impure functions in `.map()`, it's better to return both `page` and `list` from the function and then in `.subscribe()` assign them both to the properties of class.